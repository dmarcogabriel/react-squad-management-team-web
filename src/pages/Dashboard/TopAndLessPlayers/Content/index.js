import React from 'react';
import cn from 'classnames';
import { showPlayerInitials } from '../../../../utils/playersInitials';
import PlayerAvatarPlaceholder from '../../../../assets/playerAvatarPlaceholder.png';

const Content = ({ playerName, title, percentage, className, dataTestid }) => (
  <div
    data-testid={dataTestid || 'className'}
    className={cn('w-full p-5', className)}
  >
    <h2 data-testid="title" className="font-bold text-lg text-center">
      {title}
    </h2>

    <div className="relative flex justify-center my-3">
      {playerName ? (
        <div
          className={cn(
            'relative p-12 bg-white',
            'text-primary-dark rounded-full'
          )}
        >
          <p
            data-testid="playerName"
            className={cn(
              'text-3xl absolute w-full h-full',
              'flex justify-center items-center',
              'top-0 left-0 right-0'
            )}
          >
            {showPlayerInitials(playerName)}
          </p>
        </div>
      ) : (
        <div
          className={cn(
            'relative p-12 border-dashed border-2',
            'text-primary-dark rounded-full overflow-hidden'
          )}
        >
          <img
            className="absolute right-0 top-0 w-full h-full"
            src={PlayerAvatarPlaceholder}
            alt="Player Avatar"
          />
        </div>
      )}

      <p
        data-testid="percentage"
        className={cn(
          'absolute right-0 top-0',
          'text-base border-b-2 border-white pr-5'
        )}
      >
        {percentage}
      </p>
    </div>
  </div>
);

export default Content;
