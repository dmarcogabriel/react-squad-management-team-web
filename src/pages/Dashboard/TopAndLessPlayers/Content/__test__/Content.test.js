import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import Content from '..';

describe('Content tests', () => {
  afterEach(cleanup);

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Content />, div);
  });

  it('renders playerName correctly', () => {
    const { getByTestId } = render(<Content playerName="Test Name" />);

    const playerName = getByTestId('playerName');

    expect(playerName).toHaveTextContent('TN');
  });

  it('renders title correctly', () => {
    const { getByTestId } = render(<Content title="Test Title" />);

    const title = getByTestId('title');

    expect(title).toHaveTextContent('Test Title');
  });

  it('renders percentage correctly', () => {
    const { getByTestId } = render(<Content percentage="25%" />);

    const percentage = getByTestId('percentage');

    expect(percentage).toHaveTextContent('25%');
  });

  it('renders className correctly', () => {
    const { getByTestId } = render(<Content className="class-test" />);

    const className = getByTestId('className');

    expect(className).toHaveClass('class-test');
  });

  it('match snapshot', () => {
    const wrapper = shallow(
      <Content
        title="Test title"
        playerName="Test Player Name"
        percentage="25%"
        className="class-test"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
