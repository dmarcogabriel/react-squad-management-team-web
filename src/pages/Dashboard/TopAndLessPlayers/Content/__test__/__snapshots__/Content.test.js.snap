// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content tests match snapshot 1`] = `
<div
  className="w-full p-5 class-test"
  data-testid="className"
>
  <h2
    className="font-bold text-lg text-center"
    data-testid="title"
  >
    Test title
  </h2>
  <div
    className="relative flex justify-center my-3"
  >
    <div
      className="relative p-12 bg-white text-primary-dark rounded-full"
    >
      <p
        className="text-3xl absolute w-full h-full flex justify-center items-center top-0 left-0 right-0"
        data-testid="playerName"
      >
        TPN
      </p>
    </div>
    <p
      className="absolute right-0 top-0 text-base border-b-2 border-white pr-5"
      data-testid="percentage"
    >
      25%
    </p>
  </div>
</div>
`;
